File: Baekjoon/Class4/09.1149 RGB거리.java
import java.io.*;
import java.util.StringTokenizer;

class Main {
    public static void main(String[] args) throws IOException {
        int n;
        int r, g, b;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        n = Integer.parseInt(br.readLine());

        st = new StringTokenizer(br.readLine());
        r = Integer.parseInt(st.nextToken());
        g = Integer.parseInt(st.nextToken());
        b = Integer.parseInt(st.nextToken());

        for(int i = 1; i < n; i++) {
            int prevR = r, prevG = g, prevB = b;

            st = new StringTokenizer(br.readLine());
            r = Integer.parseInt(st.nextToken());
            g = Integer.parseInt(st.nextToken());
            b = Integer.parseInt(st.nextToken());
            r += Math.min(prevG, prevB); // 다른 색 중 최솟값 더하기
            g += Math.min(prevR, prevB);
            b += Math.min(prevR, prevG);
        }

        System.out.print(Math.min(Math.min(r, g), b));
        br.close();
    }
}

/* 확장 가능
import java.io.*;
import java.util.StringTokenizer;

class Main {
    private static final int COLOR_TYPES = 3;
    private static final int MAX_COST = 10000000;
    public static void main(String[] args) throws IOException {
        int n;
        int[] colors = new int[COLOR_TYPES];
        int[] prevColors = new int[COLOR_TYPES];
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        n = Integer.parseInt(br.readLine());

        st = new StringTokenizer(br.readLine());
        for(int i = 0; i < COLOR_TYPES; i++)
            colors[i] = Integer.parseInt(st.nextToken());

        for(int i = 1; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            for(int j = 0; j < COLOR_TYPES; j++) {
                prevColors[j] = colors[j];
                colors[j] = Integer.parseInt(st.nextToken());
            }

            for(int j = 0; j < COLOR_TYPES; j++) { // 다른 색 중 최솟값 더하기
                int prevMinCost = MAX_COST;
                for(int k = 0; k < COLOR_TYPES; k++)
                    if(j != k)
                        prevMinCost = Math.min(prevMinCost, prevColors[k]);
                colors[j] += prevMinCost;
            }
        }

        int totalMinCost = colors[0];
        for(int i = 1; i < COLOR_TYPES; i++)
            totalMinCost = Math.min(totalMinCost, colors[i]);
        System.out.print(totalMinCost);
        br.close();
    }
}
*/

File: Baekjoon/Class4/09.1149 RGB거리.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n;
	cin >> n;

	int r, g, b;
	cin >> r >> g >> b;

	for (int _ = 1; _ < n; _++) {
		int prev_r = r, prev_g = g, prev_b = b;

		cin >> r >> g >> b;
		r += min(prev_g, prev_b); // 다른 색 중 최솟값 더하기
		g += min(prev_r, prev_b);
		b += min(prev_r, prev_g);
	}

	cout << min(min(r, g), b);
	return 0;
}

/* 확장 가능
#include <iostream>
#include <algorithm>
using namespace std;
constexpr int COLOR_TYPES = 3;
constexpr int MAX_COST = 10000000;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n;
	cin >> n;

	int colors[COLOR_TYPES];
	for (int i = 0; i < COLOR_TYPES; i++)
		cin >> colors[i];

	for (int _ = 1; _ < n; _++) {
		int prev_colors[COLOR_TYPES];
		for (int i = 0; i < COLOR_TYPES; i++) {
			prev_colors[i] = colors[i];
			cin >> colors[i];
		}

		for (int i = 0; i < COLOR_TYPES; i++) { // 다른 색 중 최솟값 더하기
			int prev_min_cost = MAX_COST;
			for (int j = 0; j < COLOR_TYPES; j++)
				if (i != j)
					prev_min_cost = min(prev_min_cost, prev_colors[j]);
			colors[i] += prev_min_cost;
		}
	}

	int total_min_cost = colors[0];
	for (int i = 1; i < COLOR_TYPES; i++)
		total_min_cost = min(total_min_cost, colors[i]);
	cout << total_min_cost;
	return 0;
}
*/